release_dashboard/lib/widgets/progression.dart

```diff
@@ -76,6 +90,31 @@ class MainProgressionState extends State<MainProgression> {
 
   @override
   Widget build(BuildContext context) {
+    Map<ConductorStatusEntry, Object>? releaseStatus = context.watch<StatusState>().releaseStatus;
+
+    setState(() {
```

Comment:
I'm not sure why this isn't throwing an assert/error, however this looks wrong since it is updating the state during a build.

This is also wrong because here you're overwriting `_completedStep` based on the state of `releaseStatus` but then in the `.nextStep()` method you're updating it based on user interaction. What will end up happening is when a user presses a "continue" button, `nextStep` will be called, increment `_completedStep += 1`, then after that `setState` callback runs, the engine will schedule a new build of this widget, at which time you will trigger another `setState` updating `_completedStep` based on `releaseStatus`, then render the widget, then the engine will schedule a new build of the widget AGAIN.

Reply:
That's a good point.